analyzers/src/SonarAnalyzer.Common/Common/SecondaryLocation.cs

```diff
@@ -59,6 +59,6 @@ public static class SecondaryLocationHelper
             diagnostic.AdditionalLocations.Count <= index
                     ? throw new ArgumentOutOfRangeException(nameof(index))
                     : new SecondaryLocation(diagnostic.AdditionalLocations[index],
-                                            diagnostic.Properties.GetValueOrDefault(index.ToString()));
+                                            ((IDictionary<string, string>)diagnostic.Properties).GetValueOrDefault(index.ToString()));
```

Comment:
There's unfortunate clash in extension names between IDictonary and IImmutableDictionary -> penalty for overusing extensions by library makers 😞